src/snowflake/snowpark/session.py

```diff
@@ -645,9 +646,10 @@ def auto_clean_up_temp_table_enabled(self) -> bool:
         The default value is ``False``.
 
         Note:
-            Even if this parameter is ``False``, Snowpark still records temporary tables when
-            their corresponding DataFrame are garbage collected. Therefore, if you turn it on in the middle of your session or after turning it off,
-            the target temporary tables will still be cleaned up accordingly.
+            The temporary tables will only be dropped when this parameter is turned on during garbage collection. That is, if the
```

Comment:
The wording of this note is very hard to read. Can we have another pass over it?

Reply:
yes fixed. Actually here I would like clarify that only if garbage collection when this parameter is on, table will be dropped. But garbage collection might not happen once the table object is not referenced anymore. It's controlled by python. So if a user turns on this parameter, a table object is not referenced whereas garbage collection doesn't occur, then user turns it off, finally the table will not be dropped.

Reply:
The most recent writeup is way clearer. Thanks for this update! I would also add that perhaps an example for a simple script where we can show where a temp table gets created, where it becomes not referenced anymore, and the scope (following this last point) within which it may get dropped assuming (1) python garbage collection is triggered, and (2) the parameter is on.

Reply:
If there are cases where this scenario can play out because of explicitly using `cache_result` or because of other (implicit) reasons for temp table creation, then we can also have an example for each such case.

Reply:
yes, added!